File: routes/articles.js
const express = require("express");
const router = express.Router();
const articles_controller = require("../controllers/articles");

router.post("/", articles_controller.create_new_article);

router.get("/:id", articles_controller.get_article_by_id);

router.get("/", articles_controller.get_all_articles);

module.exports = router;

File: routes/articles.go
package routes

import (
	"database/sql"

	"github.com/gorilla/mux"
	"goserver.com/handlers"
)


func ArticlesRoutes(router *mux.Router, db *sql.DB) {
	router.HandleFunc("/articles", handlers.CreateArticle(db)).Methods("POST")
	router.HandleFunc("/articles", handlers.DisplayAllArticlesHandler(db)).Methods("GET")
	router.HandleFunc("/article/{id}", handlers.DisplayArticleHandler(db)).Methods("GET")
}